### PR TITLE
feat: esp32 examples and update packaging to support ESP-IDF

### PR DESCRIPTION
**Pull request by JeremyTubongbanua**
**- What I did**
- created `examples/esp32/atclient_esp32_components/` which is an example on how to import atclient into your Espressif project by copying and pasting our package directories (`atchops/`, `atclient/` and `atlogger/`) to `~/esp/esp-idf/components/`
- created `examples/esp32/atclient_esp32_static_libraries/` which is an example on how to import atclient into your Espressif project by copying and pasting our static libraries & include headers into your project.
- deleted these directories since they don't apply anymore: `packages/atclient_espidf`, `examples/atclient_esp32_source`, `examples/atclient_esp32_static_components`, and `examples/atclient_esp32_static_no_components`
- `atchops/CMakeLists.txt` now `REQUIRES` `atlogger` and `uuid4`
- `atclient/CMakeLists.txt` now `REQUIRES` `json`
- `atlogger/CMakeLists.txt` does not need to `REQUIRE` `mbedtls`
- Fixed `goto` error happening in `atclient/src/notify.c`

**- How to verify it**

- Output of running the `examples/atclient_esp32_components/` example

image

### CODE REVIEW

File: packages/atclient/src/notify.c [hunk 1]
```diff
@@ -41,6 +41,8 @@ int atclient_notify(atclient *ctx, const atclient_notify_params *params, char **
   char *cmd = NULL;
   size_t cmdsize = 0;
 
+  unsigned char *recv = NULL;
```
Comment (JeremyTubongbanua) [inline packages/atclient/src/notify.c:44]: I moved this variable here because of `goto` errors

File: packages/atclient/include/atclient/cjson.h [hunk 2]
```diff
@@ -1,6 +1,10 @@
 #ifndef ATCLIENT_CJSON_H
 #define ATCLIENT_CJSON_H
 
+#if defined(CONFIG_IDF_TARGET_ESP32)
+#include <cjson.h>
+#else
 #include <cJSON.h>
+#endif
```
Comment (JeremyTubongbanua) [inline packages/atclient/include/atclient/cjson.h:8]: `atclient/cjson.h` has a preprocessor check if we're ESP32

Then, the `atclient`'s CMakeLists.txt will now do `REQUIRES json` for cJSON stuff.

File: packages/atclient/CMakeLists.txt [hunk 3]
```diff
@@ -31,29 +31,29 @@ set(
...
 # register as an espidf component
 if(ESP_PLATFORM)
   # Run special CMake commands for ESP32
   idf_component_register(
-  SRCS ${ATCLIENT_SOURCES}
-  INCLUDE_DIRS ${ATCLIENT_INCLUDE_DIR}
-  REQUIRES mbedtls atchops
+    SRCS ${ATCLIENT_SOURCES}
+    INCLUDE_DIRS ${ATCLIENT_INCLUDE_DIR}
+    REQUIRES mbedtls atchops json
```
Comment (JeremyTubongbanua) [inline packages/atclient/CMakeLists.txt:44]: `atclient` now `REQURIES json`

File: packages/atchops/CMakeLists.txt [hunk 4]
```diff
@@ -17,17 +17,18 @@ set(
...
 cmake_policy(SET CMP0135 NEW)
 
 # register as an espidf component
-if(ESP_PLATFORM)
+if(ESP_PLATFORM)  
   idf_component_register(
         SRCS ${ATCHOPS_SOURCES}
         INCLUDE_DIRS ${ATCHOPS_INCLUDE_DIR}
-        REQUIRES mbedtls
+        REQUIRES mbedtls atlogger uuid4
```
Comment (JeremyTubongbanua) [inline packages/atchops/CMakeLists.txt:29]: `atchops` now `REQURIES mbedtls atlogger uuid4`

We will now be expecting [uuid4](https://github.com/atsign-foundation/uuid4) to be a component

File: packages/atlogger/CMakeLists.txt [hunk 5]
```diff
@@ -12,7 +12,6 @@ if(ESP_PLATFORM)
   idf_component_register(
         SRCS ${ATLOGGER_SOURCES}
         INCLUDE_DIRS ${ATLOGGER_INCLUDE_DIR}
-        REQUIRES mbedtls
```
Comment (JeremyTubongbanua) [inline packages/atlogger/CMakeLists.txt:15]: `atlogger` doesn't need mbedtls

**Comment by JeremyTubongbanua** [comment]
Setting to draft, I want to add `idf_component.yml` first

**Comment by JeremyTubongbanua** [comment]
Ready for review

**Review by realvarx — APPROVED** [review]

**Pull request merged**